lib/jellyfish_web/controllers/component_controller.ex

```diff
@@ -83,6 +83,14 @@ defmodule JellyfishWeb.ComponentController do
         {:error, :bad_request,
          "S3 credentials has to be passed either by request or at application startup as envs"}
 
+      {:error, :overridding_credentials} ->
+        {:error, :bad_request,
+         "Conflicting S3 credentials supplied via environment variables and the REST API. Please provide credentials through only one method"}
```

Comment:
It's only a suggestion, but from this message, it sounds like the user still can make some changes, e.g., remove `environment_variables`. IMO, the response should mention that this value is already set, and overriding it is prohibited/disabled.